🏗  Removed labs from settings table

naz opened the pull request:
refs https://github.com/TryGhost/Team/issues/332

By default members will be always "on" starting Ghost 4.0, so there's no need for this flag anymore. 

Note for reviewers. the migration under review here is - https://github.com/TryGhost/Ghost/pull/12639/commits/c86e674a3507f037fdce79646b017dd87600b4bf#diff-0c3b0fd661dd42a7397fbd89076b436b028481c32ea5f169b022b6508def52ae. 

The rest of changes are needed because the migration breaks quite a lot of tests after execution.

naz committed:
🏗  Removed labs from setting

refs https://github.com/TryGhost/Team/issues/332

- The last value that has been used in the code was "members"
- By default members will be always "on" starting Ghost 4.0, so there's no need for this flag anymore
- Therefore there's no real need to keep "labs" around

tpatel commented on core/server/data/migrations/versions/4.0/16-remove-labs-setting.js:
```diff
@@ -0,0 +1,23 @@
+
+    if (!result || !result[0]) {
+        logging.warn(`Could not find labs setting`);
+        result = [{}];
```
I think you should `return` here, otherwise the migration will break on `result[0].value`.

naz replied:
@tpatel I've decided the best thing here would be to completely remove the labs as a key altogether :sweat_smile:

tpatel reviewed (changes requested):
I think there could be an issue when the lab settings isn't there (although I'm not sure when it can happen). I wrote the comment inline 📝

naz committed:
Dropped support for labs setting parameter

refs https://github.com/TryGhost/Ghost/issues/10318

- `labs` setting is dropped from setting values as the use of JSON objec
to sore settings has been deprecated
- `labs` setting is no longer accepted as a paramter in the Settings API nor the
impoprter. The value is ignored if present in the POST/PUT requests and
returns 404 in case it is requested by key at `GET /settings/:key`

tpatel reviewed (approved):
Looks great 👌

naz committed:
Hardcoded labs to always return members:true

refs https://github.com/TryGhost/Ghost/issues/10318

- Because members is effectively "enabled" by default starting Ghost 4.0 have hardcoded labs setting to be such. The alternative of removing this key from labs would be equivalent to `labs.members === false` which is undesireable and would mean additional work on theme developer's side.
